src/overlays/actors/ovl_En_Time_Tag/z_en_time_tag.c

```diff
@@ -100,205 +113,224 @@ void func_80AC9FE4(EnTimeTag* this, PlayState* play) {
             ActorCutscene_SetIntentToPlay(this->actor.cutscene);
         } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
             ActorCutscene_Start(this->actor.cutscene, &this->actor);
-            this->actor.home.rot.x = 0;
+            TIMETAG_DIARY_TIMER(&this->actor) = 0;
         } else {
             ActorCutscene_SetIntentToPlay(this->actor.cutscene);
         }
     }
 }
 
-void func_80ACA5F8(EnTimeTag* this, PlayState* play) {
+void EnTimeTag_Diary_Wait(EnTimeTag* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
         if (gSaveContext.save.playerForm == PLAYER_FORM_ZORA) {
-            if (ENTIMETAG_GET_SWITCHFLAG(&this->actor) == 1) {
+            if (TIMETAG_DIARY_GET_TYPE(&this->actor) == TIMETAG_DIARY_LULU) {
                 Message_StartTextbox(play, 0x101C, &this->actor);
             } else {
+                // TIMETAG_DIARY_MIKAU
                 Message_StartTextbox(play, 0x122B, &this->actor);
             }
-            this->actor.home.rot.x = 1;
+            TIMETAG_DIARY_TIMER(&this->actor) = 1;
         } else {
+            // unable to read Zora script
             Message_StartTextbox(play, 0x122A, &this->actor);
+
+            //! FAKE:
             if (0) {}
+
             ((EnElf*)GET_PLAYER(play)->tatlActor)->unk_264 |= 4;
             Actor_ChangeFocus(&this->actor, play, GET_PLAYER(play)->tatlActor);
-            this->actor.home.rot.x = 0;
+            TIMETAG_DIARY_TIMER(&this->actor) = 0;
         }
-        this->actionFunc = func_80ACA418;
+        this->actionFunc = EnTimeTag_Diary_Cutscene;
     } else if ((this->actor.xzDistToPlayer < 100.0f) && Player_IsFacingActor(&this->actor, 0x3000, play)) {
         func_800B8614(&this->actor, play, 110.0f);
     }
 }
 
-void func_80ACA714(EnTimeTag* this, PlayState* play) {
+void EnTimeTag_KickOut_DoNothing(EnTimeTag* this, PlayState* play) {
 }
 
-void func_80ACA724(EnTimeTag* this, PlayState* play) {
+void EnTimeTag_KickOut_Transition(EnTimeTag* this, PlayState* play) {
     if (Message_GetState(&play->msgCtx) == TEXT_STATE_5) {
-        play->nextEntrance = play->setupExitList[ENTIMETAG_GET_1F(&this->actor)];
+        play->nextEntrance = play->setupExitList[TIMETAG_KICKOUT_GET_EXIT_INDEX(&this->actor)];
         play->transitionTrigger = TRANS_TRIGGER_START;
-        if (!ENTIMETAG_GET_E000(&this->actor)) {
+        if (TIMETAG_GET_TYPE(&this->actor) == TIMETAG_KICKOUT_DOOR) {
             Actor_PlaySfxAtPos(&this->actor, NA_SE_OC_DOOR_OPEN);
         }
-        this->actionFunc = func_80ACA714;
+        this->actionFunc = EnTimeTag_KickOut_DoNothing;
     }
 }
 
-void func_80ACA7C4(EnTimeTag* this, PlayState* play) {
-    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_63_01) && !CHECK_WEEKEVENTREG(WEEKEVENTREG_63_02)) {
+/**
+ * Setup a request to kick out, but wait for an event to complete first
+ */
+void EnTimeTag_KickOut_WaitForEvent(EnTimeTag* this, PlayState* play) {
```

Comment:
Should this be in the actor or on the defines for the WEEKEVENTREGS? Because that explanation is already there in 3 paragraphs, so those 3 paragraphs should be clarified there if it's not already clear to you

Reply:
It's a bit tricky because this is an unusual way to use the weekeventregs, usually they're just set once and maybe unset on cycle reset. Hence I'm a bit reluctant to bloat that header since it's massive already, seems like it belongs more in the actor that is most responsible for them? I think one source of confusion is that the flags are not used for every kickout, so if you see them by themselves and that's the only explanation, you form a misconception of the general shape of the system, hence why I think an overall explanation in the actor, cross-referenced by the flags in the header, is a better way to go.

Reply:
Would you have alternate suggestions for the names of these two `WEEKEVENTREG` as well?

Reply:
I might consider `HOLD` or `WAIT` over `BLOCK`, I think the other one has a reasonable name.

Reply:
I went with `WAIT` for now, also added a lot to the function descriptions, and just referenced those functions in save.h